plugins/permission-node/src/ServerPermissionClient.ts

```diff
@@ -103,10 +104,12 @@ export class ServerPermissionClient implements PermissionEvaluator {
     if (!token) {
       return false;
     }
-    return this.tokenManager
-      .authenticate(token)
-      .then(() => true)
-      .catch(() => false);
+    const { sub } = decodeJwt(token);
+    if (sub !== 'backstage-server') {
+      return false;
+    }
+    await this.tokenManager.authenticate(token);
```

Comment:
I think it's a good idea to throw an error in case we detect something is wrong. I am a bit concerned about the usage of `decodeJwt` above, as it seems to replicate some logic of the token manager. Perhaps, instead of decoding the token earlier, we could always catch the errors of `tokenManager.authenticate()` and re-throw them for cases where it makes sense?

Reply:
I think that makes sense, I'll rework this

Reply:
Hmm, either I'd need to tie the implementation here to specific error messages or use the pattern which would require an API change,
```
 try {
      await this.identityApi.getIdentity({
        request: { headers: { authorization: `Bearer ${token}` } } as Request,
      });
    } catch (err) {
      await this.tokenManager.authenticate(token);
    }
```

The primary error message we're looking for is 
```
{
    "error": {
        "name": "AuthenticationError",
        "message": "Invalid server token; caused by TypeError: Key for the ES256 algorithm must be one of type KeyObject or CryptoKey. Received an instance of Buffer",
        "cause": {
            "name": "TypeError",
            "message": "Key for the ES256 algorithm must be one of type KeyObject or CryptoKey. Received an instance of Buffer",
            "stack": "TypeError: Key for the ES256 algorithm must be one of type KeyObject or CryptoKey. Received an instance of Buffer\n    at asymmetricTypeCheck (/Users/sennyeya/git/backstage/node_modules/jose/dist/node/cjs/lib/check_key_type.js:17:15)\n    at checkKeyType (/Users/sennyeya/git/backstage/node_modules/jose/dist/node/cjs/lib/check_key_type.js:44:9)\n    at flattenedVerify (/Users/sennyeya/git/backstage/node_modules/jose/dist/node/cjs/jws/flattened/verify.js:79:37)\n    at compactVerify (/Users/sennyeya/git/backstage/node_modules/jose/dist/node/cjs/jws/compact/verify.js:18:60)\n    at jwtVerify (/Users/sennyeya/git/backstage/node_modules/jose/dist/node/cjs/jwt/verify.js:9:58)\n    at ServerTokenManager.authenticate (/Users/sennyeya/git/backstage/packages/backend-common/src/tokens/ServerTokenManager.ts:185:19)\n    at ServerPermissionClient.isValidServerToken (/Users/sennyeya/git/backstage/plugins/permission-node/src/ServerPermissionClient.ts:113:29)\n    at ServerPermissionClient.isEnabled (/Users/sennyeya/git/backstage/plugins/permission-node/src/ServerPermissionClient.ts:122:51)\n    at ServerPermissionClient.authorizeConditional (/Users/sennyeya/git/backstage/plugins/permission-node/src/ServerPermissionClient.ts:93:24)\n    at AuthorizedEntitiesCatalog.facets (/Users/sennyeya/git/backstage/plugins/catalog-backend/src/service/AuthorizedEntitiesCatalog.ts:269:32)\n    at <anonymous> (/Users/sennyeya/git/backstage/plugins/catalog-backend/src/service/createRouter.ts:235:48)\n    at handleReturn (/Users/sennyeya/git/backstage/node_modules/express-promise-router/lib/express-promise-router.js:24:23)\n    at /Users/sennyeya/git/backstage/node_modules/express-promise-router/lib/express-promise-router.js:64:7\n    at handleReturn (/Users/sennyeya/git/backstage/node_modules/express-promise-router/lib/express-promise-router.js:24:23)\n    at /Users/sennyeya/git/backstage/node_modules/express-promise-router/lib/express-promise-router.js:64:7\n    at Layer.handle [as handle_request] (/Users/sennyeya/git/backstage/node_modules/express/lib/router/layer.js:95:5)"
        },
        "stack": "AuthenticationError: Invalid server token; caused by TypeError: Key for the ES256 algorithm must be one of type KeyObject or CryptoKey. Received an instance of Buffer\n    at ServerTokenManager.authenticate (/Users/sennyeya/git/backstage/packages/backend-common/src/tokens/ServerTokenManager.ts:207:11)\n    at ServerPermissionClient.isValidServerToken (/Users/sennyeya/git/backstage/plugins/permission-node/src/ServerPermissionClient.ts:113:5)\n    at ServerPermissionClient.isEnabled (/Users/sennyeya/git/backstage/plugins/permission-node/src/ServerPermissionClient.ts:122:40)\n    at ServerPermissionClient.authorizeConditional (/Users/sennyeya/git/backstage/plugins/permission-node/src/ServerPermissionClient.ts:93:13)\n    at AuthorizedEntitiesCatalog.facets (/Users/sennyeya/git/backstage/plugins/catalog-backend/src/service/AuthorizedEntitiesCatalog.ts:269:7)\n    at <anonymous> (/Users/sennyeya/git/backstage/plugins/catalog-backend/src/service/createRouter.ts:235:26)"
    },
    "request": {
        "method": "GET",
        "url": "/entity-facets?facet=spec.lifecycle"
    },
    "response": {
        "statusCode": 401
    }
}
```

but it might also be possible to get to this line,
https://github.com/backstage/backstage/blob/patch/v1.22.0/packages/backend-common/src/tokens/ServerTokenManager.ts#L187-L189. That feels like we're still tying to implementation.

Reply:
@vinzscam thoughts here?